app/components/post_textbox/post_textbox_base.js

```diff
@@ -396,17 +396,68 @@ export default class PostTextBoxBase extends PureComponent {
         }
     };
 
-    handlePostDraftSelectionChanged = (event) => {
-        const cursorPosition = event.nativeEvent.selection.end;
+    handlePostDraftSelectionChanged = (event, fromHandleTextChange) => {
+        let cursorPosition;
+
+        // When called from PasteableTextInput's onSelectionChange, nativeEvent gets passsed with
+        // updated cursorPosition. When called fromHandleTextChange, there's no nativeEvent passed but
+        // cursorPosition is already updated and available in state.
+        if (fromHandleTextChange) {
+            cursorPosition = this.state.cursorPosition;
+        } else {
+            cursorPosition = event.nativeEvent.selection.end;
+        }
+
         const {cursorPositionEvent} = this.props;
 
         if (cursorPositionEvent) {
             EventEmitter.emit(cursorPositionEvent, cursorPosition);
         }
 
-        this.setState({
-            cursorPosition,
-        });
+        // Workaround to avoid iOS emdash autocorrect in Code Blocks
+        if (Platform.OS === 'ios') {
+            // Matches all text between a pair of triple backticks ```, inclusive
+            // OR all text after a dangling triple backtick ```, inclusive
+            // Triple backticks must also be on their own separate line to be matched
+            const regexForCodeBlock = /^```$(.*?)^```$|^```$(.*)/gms;
+
+            // Loop through this.state.value and create array of all regex match positions, start and end
+            // exec() method details: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/exec
+            const matches = [];
+            let match;
+            while ((match = regexForCodeBlock.exec(this.state.value)) !== null) {
+                matches.push({
+                    start: regexForCodeBlock.lastIndex - match[0].length, // start position of a regex match
+                    end: fromHandleTextChange ? regexForCodeBlock.lastIndex : regexForCodeBlock.lastIndex + 1, // end position of a regex match
+                });
```

Comment:
Could this be a `.map()` on `regexForCodeBlock.exec(this.state.value)` to build out the `matches` object array.

Reply:
I revised this to a cleaner version with `.map()` and `string.matchAll(regExp)`. However, turns out `matchAll` is only supported in node12 and up as shown in the now failed test.

`regExp.exec(string)` requires a loop to build out the object array since its output isn't actually an array with all matches so we can't run `map()` directly on it. 

I'm unsure of the importance of the node environment relative to this so I defer to you here for advice.

Reply:
I see. Let's go back to what you had before (using `.exec`) and manually creating the array of cursor position ranges.

Now that we have this function in two places, it should move into a single spot (perhaps `app/utils/markdown`) and with unit tests. That way, we can safely tweak this logic of cursor position range checking (and even the regex, if necessary) while ensuring we don't change the desired/expected behaviour.

Another option that came to mind was along the lines of:
```
    const matches = value.match(regexForCodeBlock).map(match => {
      return {
        start: //start logic
        end: // end logic
      }
    });

```
but let's wrap the current functionality in tests first before changing anything.